doc/developer-notes.md

```diff
@@ -839,6 +840,37 @@ namespace {
 #endif // BITCOIN_FOO_BAR_H
 ```
 
+Construct On First Use
+------------------------
+
+It is not possible to control the order in which non-local static objects in *different* translation units are initialized.
+This is the so-called "static initialization order problem". It could cause the use of an object:
+- before it is constructed (on program startup)
+- after it is destructed (at program exit)
+
+Both cases lead to undefined behavior.
+
+To prevent the static initialization order problem, use the Construct On First Use:
+
+```C++
+BCLog::Logger& LogInstance()
+{
+    // See doc/developer-notes.md#construct-on-first-use
+    static BCLog::Logger* g_logger{new BCLog::Logger()};
+    return *g_logger;
+}
+```
+
+*Rationale*: This approache guarantees that:
+
+- an object is constructed prior to its first use
```

Comment:
sorry to re-review the same area, but I think this could be reduced to two points and would be better written as:
- the object is constructed before first use and not destroyed until after last use
- the object initialization is thread-safe

(also: destructed -> destroyed and the "its" can be removed)

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/18881#issuecomment-624053939).

Reply:
"its" seems prevent any ambiguation :)